### ISSUE TITLE
Lbry.tv and Odysee.com engines

### ISSUE THREAD

**Issue by nukelr** [searx/searx#2504]
**Working URL to the engine**
https://lighthouse.lbry.com/search?s=QUERYSTRING&size=5&from=0&nsfw=false
**Why do you want to add this engine?**
_First and foremost, LBRY is a new protocol that allows anyone to build apps that interact with digital content on the LBRY network. Apps built using the protocol allow creators to upload their work to the LBRY network of hosts (like BitTorrent), to set a price per stream or download (like iTunes) or give it away for free (like YouTube without ads). The work you publish could be videos, audio files, documents, or any other type of file.
Traditional video (or other content) sites such as YouTube, Instagram, and Spotify store your uploads on their servers and allow viewers to download them. They also allow creators to make some money through advertising or other mechanisms. However, there are some well-known drawbacks, especially for people whose material is perceived as not being advertiser-friendly.
LBRY aims to be an alternative to these sites, allowing publishers and their fans to interact directly without the risk of demonetization or other meddling._

**Features of this engine**
LBRY indexes digital content using an open blockchain

**How can Searx fetch the information from this engine?**
https://lighthouse.lbry.com/search?s=youtube-is-over-lbry-odysee-are-here&size=5&from=0&nsfw=false
**Applicable category of this engine**
Mostly Videos but lbry has all kind of content (Docs, pictures....)

**Comment by ralyodio** [comment]
yes, this would be great for videos. a lot of people are leaving youtube and uploading to odysee.

**Comment by ghost** [comment]
I think a LBRY/odysee search would be fantastic.

**Comment by unixfox** [comment]
@craig9 Feel free to contribute into the Searx project if you want to get a faster integration of Lbry into it :).

**Comment by ghost** [comment]
@unixfox Thank you, I will consider it.

**Comment by ghost** [comment]
I did some poking around in the lighthouse code (the LBRY search engine), and found a semi-hidden page of documentation which gives access to more search parameters, which will make getting the info for a searx result page a bit easier. Here is the page with the additional search params:

https://github.com/lbryio/lighthouse/tree/master/app/actions/search 

And here is a test URL which also gives channel name, and a thumbnail_url, and restricts results to video only as an example

https://lighthouse.lbry.com/search?s=test&include=channel,thumbnail_url&mediaType=video

I may look further into this yet.

**Comment by 0xhtml** [comment]
I took a look and I would extend the URL to this:
https://lighthouse.lbry.com/search?s=test&include=channel,thumbnail_url,title,description,duration,release_time&mediaType=video

Example result:
```json
{
  "channel": "@InfoToons",
  "claimId": "69e377fcaf7acb6b812496b6983e9820ace48a27",
  "description": "This is a test animation of the Salty Cracker character I modeled and rigged in Blender over the weekend.",
  "duration": 21,
  "name": "Salty-Test-Animation-B720p0001-0524",
  "release_time": "2021-03-01T12:05:09Z",
  "thumbnail_url": "https://spee.ch/0/ba6d4c72a0805155.png",
  "title": "Salty Cracker (Test Animation)"
},
```

- When `nsfw=false` is passed, NSFW content is filtered out. (for Safe Search: Moderate/Strict)
- Language filtering is possible with `language=...`. I couldn't find a list with supported languages but `de` and `en` as well as `de-DE` and `en-US` seem to work.
- Paging is possible with `size=10&from=...`. `from` being the offset (e.g. 10 * page number)
- Specifying a time range doesn't seem to be supported.

**Comment by Finoway** [comment]
> I took a look and I would extend the URL to this:
> https://lighthouse.lbry.com/search?s=test&include=channel,thumbnail_url,title,description,duration,release_time&mediaType=video
> 
> Example result:
> 
> ```json
> {
>   "channel": "@InfoToons",
>   "claimId": "69e377fcaf7acb6b812496b6983e9820ace48a27",
>   "description": "This is a test animation of the Salty Cracker character I modeled and rigged in Blender over the weekend.",
>   "duration": 21,
>   "name": "Salty-Test-Animation-B720p0001-0524",
>   "release_time": "2021-03-01T12:05:09Z",
>   "thumbnail_url": "https://spee.ch/0/ba6d4c72a0805155.png",
>   "title": "Salty Cracker (Test Animation)"
> },
> ```
> 
>     * When `nsfw=false` is passed, NSFW content is filtered out. (for Safe Search: Moderate/Strict)
> 
>     * Language filtering is possible with `language=...`. I couldn't find a list with supported languages but `de` and `en` as well as `de-DE` and `en-US` seem to work.
> 
>     * Paging is possible with `size=10&from=...`. `from` being the offset (e.g. 10 * page number)
> 
>     * Specifying a time range doesn't seem to be supported.

Hello, do you think it will be added please ?

**Comment by henk717** [comment]
In case it hangs on this issue since I have not seen it discussed, you can resolve this data to a working link as follows.
Begin with https://odysee.com for the result, then add /name:claimId
For example we can use the following video : https://odysee.com/@Luke:7/the-proper-way-to-do-web-searches-searx:3
If we use the above mentioned method to do a search like this : https://lighthouse.lbry.com/search?s=the-proper-way-to-do-web-searches-searx&include=channel,thumbnail_url,title,description,duration,release_time&mediaType=video

You will see that the name is the-proper-way-to-do-web-searches-searx and the claimID is 3f09985f052b14e0fc5367c9dfd2554bffbf9869, which would make the result https://odysee.com/the-proper-way-to-do-web-searches-searx:3f09985f052b14e0fc5367c9dfd2554bffbf9869 which corresponds to the same video. This is the same format they'd use for the linkbacks from their embedded player with the omission of ?src=embed .
